devices/Sim7080/Sim7080G.cs

```diff
@@ -0,0 +1,301 @@
+        /// <param name="enableReporting">Report the network system mode information</param>
+        /// <param name="wait">The time to wait to switch system mode</param>
+        public void SetNetworkSystemMode(SystemMode systemMode = SystemMode.GSM, bool enableReporting = true, int wait = 5000)
+        {
+            SimController.SetSystemMode(_serialPort, systemMode, enableReporting);
+
+            Thread.Sleep(wait);
+        }
+
+        /// <summary>
+        /// Connect to the cellular network
+        /// </summary>
+        //// <param name="apn">Cellular network access point name</param>
+        public void NetworkConnect(string apn)
+        {
+            if (!_serialPort.IsOpen)
+            {
+                NetworkConnected = ConnectionStatus.Disconnected;
+                return;
+            }
+
+            var retryCount = 0;
+
+            do
+            {
+                retryCount++;
+
+                SimController.NetworkConnect(_serialPort, apn);
+
+            } while (NetworkConnected == ConnectionStatus.Disconnected && retryCount < Retry);
+        }
+
+        /// <summary>
+        /// Disconnect to the cellular network 
+        /// </summary>
+        public void NetworkDisconnect()
+        {
+            if (_serialPort.IsOpen &&
+                NetworkConnected == ConnectionStatus.Connected)
+            {
+                SimController.NetworkDisconnect(_serialPort);
+            }
+        }
+
+        /// <summary>
+        /// Connect to Azure IoT Hub using MQTT protocol
+        /// </summary>
+        /// <param name="deviceId">The device identifier</param>
+        /// <param name="hubName">The Azure IoT Hub Name</param>
+        /// <param name="sasToken">The Secure Access Token</param>
+        /// <param name="portNumber">The MQTT port number</param>
+        /// <param name="apiVersion">The Azure IoT Hub API version</param>
+        /// <param name="wait">The time to wait to establish the connection</param>
+        public void ConnectAzureIoTHub(string deviceId, string hubName, string sasToken, int portNumber = 8883, string apiVersion = "2021-04-12", int wait = 5000)
+        {
+            if (!_serialPort.IsOpen || NetworkConnected == ConnectionStatus.Disconnected)
+            {
+                EndpointConnected = ConnectionStatus.Disconnected;
+                return;
+            }
+
+            string username = $"{hubName}.azure-devices.net/{deviceId}/?api-version={apiVersion}";
+            string endpointUrl = $"{hubName}.azure-devices.net";
+
+            var retryCount = 0;
+
+            do
+            {
+                retryCount++;
+
+                EndpointConnected = SimController.EndpointConnect(_serialPort, deviceId, endpointUrl, portNumber, username, sasToken, wait);
+
+            } while (EndpointConnected == ConnectionStatus.Disconnected && retryCount < Retry);
+
+            //https://learn.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-c2d
```

Comment:
this is a bit more complex. See the Azure Device constructor: https://github.com/nanoframework/nanoFramework.Azure.Devices/blob/401f20aa6ca95223c67c017b9cd1d642e00d8cf3/nanoFramework.Azure.Devices.Client/DeviceClient.cs#L74